packages/relay/src/services/ReportService.ts

```diff
@@ -0,0 +1,78 @@
+import { DB } from 'anondb'
+import { nanoid } from 'nanoid'
+import { Groth16Proof, PublicSignals } from 'snarkjs'
+import { commentService } from '../services/CommentService'
+import { postService } from '../services/PostService'
+import {
+    CommentNotExistError,
+    InvalidPostIdError,
+} from '../types/InternalError'
+import { ReportHistory, ReportType } from '../types/Report'
+import { UnirepSocialSynchronizer } from './singletons/UnirepSocialSynchronizer'
+import ProofHelper from './utils/ProofHelper'
+
+export class ReportService {
+    async verifyReportData(
+        db: DB,
+        reportData: ReportHistory,
+        publicSignals: PublicSignals,
+        proof: Groth16Proof,
+        synchronizer: UnirepSocialSynchronizer
+    ): Promise<ReportHistory> {
+        // 1.a Check if the post / comment exists is not reported already(post status = 1 / comment status = 1)
+        if (reportData.type === ReportType.Post) {
+            const post = await postService.fetchSinglePost(
```

Comment:
After modifying fetchSinglePost logic, we don't sent status to fetch the single post. In this way, the PostIsReportedError should be thrown if the status is equal to PostStatus.Reported( 2 )